src/backend/distributed/connection/remote_commands.c

```diff
@@ -616,6 +628,8 @@ PutRemoteCopyEnd(MultiConnection *connection, const char *errormsg)
 
 	/* see PutRemoteCopyData() */
 
+	connection->copyBytesWrittenSinceLastFlush = 0;
```

Comment:
We keep the connection open after an abort and currently we don't call `PutRemoteCopyEnd` prior to closing a connection (which causes "unexpected message type 0x58" errors, but that's a separate issue). 

We should probably reset this flag in `AfterXactHostConnectionHandling`.

Reply:
Thinking about this more, it's probably ok since we don't cache connections if they are still in a transaction when the abort happens. Worth double checking.

Reply:
From what I‌ understand,
1. If the transaction is rolled back after a *successful* `COPY FROM`, `PutRemoteCopyEnd()` has already been called, and we are good.
2. If the transaction is rolled back after an *unsuccessful* `COPY FROM`, `PQstatus()` returns `CONNECTION_BAD` so we shut-down and free the connection in `AfterXactHostConnectionHandling()`. So we are good in this case too, since we will use a new connection for the next copy.

But I think resetting copy state in `AfterXactHostConnectionHandling()` doesn't hurt, so I added it.

Also, please note that the only place that we have to reset `copyBytesWrittenSinceLastFlush` is `PutRemoteCopyData()`. Other places are mostly nice-to-haves and shouldn't impact the performance much, and doesn't impact correctness at all.